File: 1101-1200/1111-Maximum Nesting Depth of Two Valid Parentheses Strings/1111-Maximum Nesting Depth of Two Valid Parentheses Strings.cpp
class Solution {
public:
    vector<int> maxDepthAfterSplit(string seq) {
        int A = 0, B = 0;
        vector<int> result(seq.size());
        for (int i = 0; i < seq.size(); ++i) {
            if (seq[i] == '(') {
                if (A < B) {
                    ++A;
                }
                else {
                    ++B;
                    result[i] = 1;
                }
            }
            else {
                if (A > B) {
                    --A;
                }
                else {
                    --B;
                    result[i] = 1;
                }
            }
        }

        return result;
    }
};

File: 1101-1200/1111-Maximum Nesting Depth of Two Valid Parentheses Strings/1111-Maximum Nesting Depth of Two Valid Parentheses Strings.py
class Solution:
    def maxDepthAfterSplit(self, seq: str) -> List[int]:
        A = B = 0
        result = [0] * len(seq)
        for i, c in enumerate(seq):
            if c == '(':
                if A < B:
                    A += 1
                else:
                    B += 1
                    result[i] = 1
            else:
                if A > B:
                    A -= 1
                else:
                    B -= 1
                    result[i] = 1
        return result
